src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -574,9 +569,9 @@ private async Task InitializeAsync(bool ignoreFailures, IEnumerable<Configuratio
 
                 try
                 {
-                    SetData(await PrepareData(data, cancellationToken).ConfigureAwait(false));
-                    _applicationSettings = data;
                     _watchedSettings = watchedSettings;
+                    _mappedData = await MapConfigurationSettings(data).ConfigureAwait(false);

```

Comment:
What's the reasoning for the order switch? Initialization hasn't really finished until SetData has executed successfully, and `_mappedData` is used as the indicator for successful initialization.

Reply:
I think I don't fully understand the reason for the ordering, which is why I put it last to save a couple extra lines of code. I can put setdata first again